### PR TITLE
Fix YOLO aspect-fit preprocessing

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary
- replace Vision scale-fill preprocessing with task-specific scaleFit for localization and centerCrop for classification
- map boxes, keypoints, OBBs, and segmentation masks back through Ultralytics-style letterbox gain/padding
- keep camera preview and inference buffers in interface orientation so landscape frames reach the model right-side-up

## Validation
- git diff --check
- swiftc -typecheck -target arm64-apple-ios16.0-simulator -sdk "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.4.sdk" -I .build/arm64-apple-ios-simulator/debug/Modules Sources/YOLO/*.swift

## Notes
- Full xcodebuild simulator validation is blocked locally because this Xcode installation reports iOS 26.4 as not installed.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📱 This PR improves the Ultralytics YOLO iOS app’s accuracy and consistency by fixing how model outputs are mapped back to the screen, refining orientation handling, and updating docs, model export guidance, and app store links.

### 📊 Key Changes
- 🧭 **Fixed coordinate mapping for YOLO outputs**
  - Added shared letterbox-aware geometry conversion in `BasePredictor`.
  - Detection, segmentation, pose, and OBB results now correctly map from model space back to the original image space.
- 🖼️ **Improved preprocessing behavior by task**
  - Default Vision preprocessing now uses `scaleFit` for localization tasks.
  - Classification overrides this with `centerCrop`, which better matches classification model expectations.
- 🔲 **Corrected OBB rendering**
  - OBB boxes are now transformed into input-image space before display.
  - Polygon drawing logic was simplified so angles are applied directly in pixel space.
- 🎭 **Improved segmentation mask cropping**
  - Added logic to crop masks based on preprocessing padding.
  - This avoids masks being stretched or misaligned when letterboxing is used.
- 🤸 **Improved pose keypoint placement**
  - Pose boxes and keypoints now use the same corrected mapping pipeline as detections.
- 📹 **Better camera orientation handling**
  - `VideoCapture` now works with `AVCaptureVideoOrientation` directly.
  - Preview and inference buffer orientation are kept in sync more reliably, including when the view appears.
- 📚 **Documentation refresh**
  - Updated README files in English and Chinese.
  - Replaced older App Store links with the current Ultralytics YOLO app link.
  - Clarified wording from “detection” to broader “inference” where appropriate.
  - Updated Core ML export examples to recommend square export sizes for mixed portrait/landscape usage, with task-specific guidance for YOLO26 and OBB models.
- ✅ **Added tests for geometry conversion**
  - New unit tests cover letterbox mapping, non-square models, OBB conversion, mask crop regions, and invalid geometry edge cases.
- 🚀 **Version bump**
  - App version updated from `8.8.4` to `8.8.5`, with build number increased from `569` to `572`.

### 🎯 Purpose & Impact
- 🎯 **More accurate on-screen results**
  - Users should see boxes, masks, keypoints, and OBB overlays align more closely with the actual objects in camera and image inputs.
- 📱 **Better support across portrait and landscape**
  - The app should behave more consistently when device orientation changes or when using square versus non-square exported models.
- 🧠 **Better defaults for different task types**
  - Classification and localization tasks now use preprocessing that better matches how their models are meant to run.
- 🛠️ **Easier model export and testing**
  - Updated docs make it clearer which Core ML export sizes to use depending on task and orientation needs.
- 🔒 **Higher confidence in correctness**
  - Added tests reduce the risk of future regressions in coordinate and mask handling.
- 🌍 **Improved usability and discoverability**
  - Updated App Store links and clearer README messaging make it easier for new users to find, install, and understand the app.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/yolo-ios-app` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below to help ensure a smooth integration. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/yolo-ios-app/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/yolo-ios-app` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: Sources/YOLO/ObjectDetector.swift [hunk 1]
```diff
@@ -59,16 +59,27 @@ public final class ObjectDetector: BasePredictor, @unchecked Sendable {
         let invertedBox = CGRect(
           x: prediction.boundingBox.minX, y: 1 - prediction.boundingBox.maxY,
           width: prediction.boundingBox.width, height: prediction.boundingBox.height)
-        let imageRect = VNImageRectForNormalizedRect(
-          invertedBox, Int(inputSize.width), Int(inputSize.height))
+        let imageRect: CGRect
+        if modelInputSize.width > 0, modelInputSize.height > 0 {
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/ObjectDetector.swift]: ⚠️ **HIGH**: `VNRecognizedObjectObservation.boundingBox` is already reported in the source image's normalized coordinate space. Converting it to `modelInputSize` and then running it through `inputRect(fromModelRect:)` applies the letterbox inverse a second time, which will shift/shrink boxes for NMS-pipelined models on non-square inputs. The aspect-fit remap should stay limited to raw tensor outputs.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Nice cleanup overall: the shared letterbox helpers make the coordinate handling much easier to follow, and the pose/OBB remapping changes look directionally correct. The two issues above are worth fixing before merge: the Vision object-observation path appears to be double-correcting boxes, and segmentation now returns mixed coordinate spaces between `combinedMask` and the raw `masks` array.

💬 Posted 2 inline comments

File: Sources/YOLO/Segmenter.swift [hunk 2]
```diff
@@ -52,10 +52,15 @@ public final class Segmenter: BasePredictor, @unchecked Sendable {
...
+      let maskImage: CGImage? = self?.inputMask(
+        fromModelMask: processed.0,
+        inputSize: capturedInputSize,
+        modelInputSize: capturedModelInputSize
+      ) ?? processed.0
       let result = YOLOResult(
         orig_shape: capturedInputSize,
         boxes: boxes,
-        masks: Masks(masks: processed.1, combinedMask: processed.0),
+        masks: Masks(masks: processed.1, combinedMask: maskImage),
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/Segmenter.swift]: 💡 **MEDIUM**: Only `combinedMask` is remapped back out of the letterboxed model space here; `processed.1` is still returned unchanged. That means `result.masks?.masks` remains in padded/model coordinates while `boxes` and `combinedMask` are in input-image coordinates, which will break consumers that read the raw mask arrays. Please crop/remap the per-instance float masks as well, or keep all mask outputs in the same coordinate system.

File: Sources/YOLO/ObjectDetector.swift (hunk 1 above)
Comment (glenn-jocher) [reply Sources/YOLO/ObjectDetector.swift]: Fixed in 93acddd: NMS-pipelined VNRecognizedObjectObservation boxes now stay in Vision source-image normalized coordinates; the letterbox inverse is only used for raw tensor outputs.

File: Sources/YOLO/Segmenter.swift (hunk 2 above)
Comment (glenn-jocher) [reply Sources/YOLO/Segmenter.swift]: Fixed in 93acddd: the segmentation mask generator now applies the same crop rect to both combinedMask and the raw per-instance masks, so both outputs share the same unpadded coordinate space.

**Comment by UltralyticsAssistant** [comment]
🎉 Excited to see this merged, @glenn-jocher!

“Accuracy is the twin brother of honesty; inaccuracy, of dishonesty.” — Nathaniel Hawthorne

This update really reflects that idea: tighter letterbox-aware mapping, improved orientation handling, better task-specific preprocessing, stronger docs, and added geometry tests all help the iOS app deliver more reliable, consistent YOLO inference on real devices. The result is a better experience for users and a stronger foundation for future improvements.

Thanks for driving such a thoughtful and high-impact upgrade to the Ultralytics YOLO iOS app 🚀
